src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-sync-policy-form/rgw-multisite-sync-policy-form.component.ts

```diff
@@ -0,0 +1,175 @@
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { RgwMultisiteSyncPolicyStatus } from '../models/rgw-multisite';
+import { CdForm } from '~/app/shared/forms/cd-form';
+import _ from 'lodash';
+
+@Component({
+  selector: 'cd-rgw-multisite-sync-policy-form',
+  templateUrl: './rgw-multisite-sync-policy-form.component.html',
+  styleUrls: ['./rgw-multisite-sync-policy-form.component.scss']
+})
+export class RgwMultisiteSyncPolicyFormComponent extends CdForm implements OnInit {
+  syncPolicyForm: CdFormGroup;
+  editing = false;
+  action: string;
+  resource: string;
+  syncPolicyStatus = RgwMultisiteSyncPolicyStatus;
+
+  bucketDataSource = (text$: Observable<string>) => {
+    return text$.pipe(
+      debounceTime(200),
+      distinctUntilChanged(),
+      mergeMap((token: string) => this.getBucketTypeahead(token))
+    );
+  };
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    public actionLabels: ActionLabelsI18n,
+    private fb: CdFormBuilder,
+    private rgwMultisiteService: RgwMultisiteService,
+    private notificationService: NotificationService,
+    private rgwBucketService: RgwBucketService
+  ) {
+    super();
+    this.editing = this.router.url.startsWith(`/rgw/multisite/sync-policy/${URLVerbs.EDIT}`);
+    this.action = this.editing ? this.actionLabels.EDIT : this.actionLabels.CREATE;
+    this.resource = $localize`Sync Policy Group`;
+    this.createForm();
+    this.loadingReady();
+  }
+
+  ngOnInit(): void {
+    if (this.editing) {
+      this.route.paramMap.subscribe((params) => {
+        const groupName = params.get('groupName');
+        if (groupName) {
+          this.loadingStart();
+          this.rgwMultisiteService
+            .getSyncPolicy('', '', true)
+            .subscribe((syncPolicies: Array<Object>) => {
+              this.loadingReady();
+              if (syncPolicies && syncPolicies.length > 0) {
+                const syncPolicyToEdit: any = syncPolicies.find(
+                  (policy) => policy['id'] === decodeURIComponent(groupName)
+                );
+                if (syncPolicyToEdit) {
+                  this.syncPolicyForm.patchValue({
+                    group_id: syncPolicyToEdit.id,
+                    status: syncPolicyToEdit.status,
+                    bucket_name: syncPolicyToEdit.bucketName
+                  });
+                } else {
+                  this.goToListView();
+                }
+              }
+            });
+        }
+      });
+    }
+  }
+
+  createForm() {
+    this.syncPolicyForm = this.fb.group({
+      group_id: ['', Validators.required],
+      status: [`${this.syncPolicyStatus.ENABLED}`, Validators.required],
+      bucket_name: ['', , this.bucketExistence(true)]
+    });
+  }
+
+  goToListView() {
+    // passing state in order to return to same tab on details page
+    this.router.navigate(['/rgw/multisite'], { state: { activeId: 'syncPolicy' } });
```

Comment:
Ideally the tabs should have been routed. I [created a PR](https://github.com/ceph/ceph/pull/53207/files#diff-c23f0e8b0a31e5da270b339edb10956551fece0d5bf79a83e6c38e28f9d72e2a) long ago to make a generic routable tabs but I haven't went back to that PR in a while due to other priorities.. But maintaining the state is good for now.

Reply:
maybe I can do the routed tabs in my next PR, if it is okay!

Reply:
yup, more than fine with me. thanks